python/whylogs/extras/image_metric.py

```diff
@@ -118,15 +140,28 @@ def columnar_update(self, view: PreprocessedColumn) -> OperationResult:
             if isinstance(image, ImageType):
                 metadata = get_pil_image_metadata(image)
                 for attr in metadata.keys():
+                    if attr not in submetric_names:  # EXIF tag discovery
+                        if _trackable_exif_tag(attr, self._allowed_exif_tags, self._forbidden_exif_tags):
+                            # TODO: add a resolver to ImageMetricConfig
+                            if isinstance(metadata[attr], int) or isinstance(metadata[attr], float):
+                                self.submetrics[attr] = DistributionMetric.zero(MetricConfig())
+                            elif isinstance(metadata[attr], str):
+                                self.submetrics[attr] = FrequentItemsMetric.zero(MetricConfig())
+
                     if attr in submetric_names:
                         self.submetrics[attr].columnar_update(PreprocessedColumn.apply([metadata[attr]]))
+
                 count += 1
         return OperationResult.ok(count)
 
     @classmethod
     def zero(cls, config: MetricConfig) -> "ImageMetric":
+        if not isinstance(config, ImageMetricConfig):
+            logger.warning("ImageMetric.zero() needs an ImageMetricConfig")
+            config = ImageMetricConfig()
+
         dummy_image = new_image("HSV", (1, 1))
-        metadata = get_pil_image_metadata(dummy_image)
+        metadata = get_pil_image_metadata(dummy_image)  # any EXIF tags will be discovered as images are logged
```

Comment:
Wait read this again, why are you putting a dummy image through?

Reply:
It gives a nice SPOT for the image stats we track for all images (versus the data-dependent EXIF tags we discover as images are logged).